flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/SchedulingTopology.java

```diff
@@ -43,11 +43,27 @@
 	 */
 	Optional<SchedulingExecutionVertex> getVertex(ExecutionVertexID executionVertexId);
 
+	/**
+	 * Looks up the {@link SchedulingExecutionVertex} for the given {@link ExecutionVertexID}.
+	 *
+	 * @param executionVertexId identifying the respective scheduling vertex
+	 * @return The respective scheduling vertex or throw {@link Exception} if the vertex does not exist
+	 */
+	SchedulingExecutionVertex getVertexOrThrow(ExecutionVertexID executionVertexId);
```

Comment:
Did you consider using a default method? https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html

```
default SchedulingExecutionVertex getVertexOrThrow(ExecutionVertexID executionVertexId) {
...
}

(This is not a review)
```

Reply:
Emm, `default` is a good way to add new method in interfaces, since in some user cases it is not easy to add corresponding method in the existing implementations. But using `default` may have the risk of `existing implementations of an interface may compile without error or warning but fail at runtime` (as Item 21 in Effective Java), thus I prefer not to use `default` as far as possible.
Anyway, thanks a lot for providing this `default` proposal.

Reply:
Emm, it seems there is no risk of current implementation. We could use `default`.